mod_wsbridge/mod_wsbridge.c

```diff
@@ -384,6 +417,96 @@ websocket_write_back(struct lws *wsi_in, enum lws_write_protocol type, char *str
 	return n;
 }
 
+void add_content_type(private_t *tech_pvt, cJSON* json) {
+	if (cJSON_GetObjectItem(json, "content-type")) {
+		cJSON_DeleteItemFromObject(json, "content-type");
+	}
+	cJSON_AddItemToObject(json, "content-type", cJSON_CreateString(tech_pvt->content_type));
+}
+
+int is_mute_event(char* event, char* method) {
+	return !strcmp(event, "websocket:media:update") && !strcmp(method, "update");
+}
+
+void reset_ws_write_indexes(private_t *tech_pvt) {
+	switch_mutex_lock(tech_pvt->write_mutex);
+	tech_pvt->write_count = 0;
+	tech_pvt->write_start = 0;
+	tech_pvt->write_end = 0;
+	switch_mutex_unlock(tech_pvt->write_mutex);
+}
+
+void set_audio_active(private_t *tech_pvt, switch_bool_t value) {
+	switch_mutex_lock(tech_pvt->audio_active_mutex);
+	switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "Setting channel status. active=[%d]\n", value);
+	tech_pvt->audio_active = value;
+	switch_mutex_unlock(tech_pvt->audio_active_mutex);
+}
+
+switch_bool_t get_audio_active(private_t *tech_pvt) {
+	switch_bool_t value;
+	switch_mutex_lock(tech_pvt->audio_active_mutex);
+	value = tech_pvt->audio_active;
+	switch_mutex_unlock(tech_pvt->audio_active_mutex);
+	return value;
+}
+
+char* get_valuestring(cJSON* json, char* field) {
+	char* valuestring = NULL;
+	cJSON* json_field = NULL;
+	json_field = cJSON_GetObjectItem(json, field);
+	if (json_field) {
+		valuestring = json_field->valuestring;
+	}
+	return valuestring;
+}
+
+void on_event(private_t *tech_pvt, cJSON* json) {
+	char* event = NULL;
+	char* method = NULL;
+	event = get_valuestring(json, "event");
+	method =  get_valuestring(json, "method");
+
+	if (!event || !method) {
+		return;
+	}
+
+	if (is_mute_event(event, method)) {
+		cJSON* active;
+		active = cJSON_GetObjectItem(json, "active");
+		if (active && (active->type == cJSON_False || active->type == cJSON_True)) {
+			set_audio_active(tech_pvt, active->valueint);
+			if(!get_audio_active(tech_pvt)) {
```

Comment:
Calling get_audio_active is not necessary.  You can use a local variable to store the audio_active state and it will not be modified by another thread. You can do the following.

```
switch_bool_t audio_active = active->valueint;
set_audio_active(tech_pvt, audio_active);
if(!audio_active) {
  reset_ws_write_indexes(tech_pvt);
}

```